File: public/server/javascript/src/usr/testThrow.ts
import { Dict, DBOperate } from "../libs/lockvalserver"


function doSomething(input: DBOperate<any>) {
  input.Throw(2,"Nothing changed") // In any case, just call Throw, the call will be terminated
}


export function main(input: DBOperate<any>) {
  input.GetSubVal(input.UID, "mBase", "Count")
  input.GetAndLock()

  let c = input.GetResp.IDKey[input.UID]?.KeySub["mBase"]?.SubVal["Count"]

  let ci = Number(c)
  ci++
  c = ci.toString()

  input.PutSubVal(input.UID, "mBase", "Count", c)

  doSomething(input)

  input.PutAndUnlock()

  return {}
}

File: public/server/lua/src/usr/testThrow.lua
return require("umd").define({
    "exports",
    "other/helper",
}, function(exports, helper)

    local function doSomething(input)
        input.Throw(2,"Nothing changed") -- In any case, just call Throw, the call will be terminated
    end

    function exports.main(input)

        input.GetSubVal(input.UID, "mBase", "Count")
        input.GetAndLock()

        local c = helper.GetResp(input, input.UID, "mBase", "Count")
        if c == "" then
            c = "0"
        end
        local ci = tonumber(c)
        ci = ci + 1
        c = tostring(ci)

        input.PutSubVal(input.UID, "mBase", "Count", c)

        doSomething(input)

        input.PutAndUnlock()

        local retval = {}
        retval.Hello = "Lua"
        return retval
    end

end)
